### PR TITLE
[dynamo][fsdp] Track FSDPNNModuleVariable for mutations

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125340
* #128336

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng

### CODE REVIEW

File: test/distributed/test_dynamo_distributed.py [hunk 1]
```diff
@@ -483,6 +515,20 @@ def test_fsdp_aot_eager(self):
             outputs = fsdp_m(inputs)
             self.assertTrue(same(correct_outputs, outputs))
 
+    @skip_if_lt_x_gpu(1)
+    @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
+    @unittest.skip(
+        "Graph break not supported - see test_custom_getattr_on_module_exception"
+    )
+    def test_setattr(self):
```
Comment (anijain2305) [inline test/distributed/test_dynamo_distributed.py:523]: This test repros the graph breaks seen for Dtensor enablement.

File: test/distributed/test_dynamo_distributed.py [hunk 2]
```diff
@@ -483,6 +515,20 @@ def test_fsdp_aot_eager(self):
             outputs = fsdp_m(inputs)
             self.assertTrue(same(correct_outputs, outputs))
 
+    @skip_if_lt_x_gpu(1)
+    @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
+    @unittest.skip(
```
Comment (anijain2305) [inline test/distributed/test_dynamo_distributed.py:520]: Deliberate .. I don't have good ideas on how to support this.

**Comment by anijain2305** [comment]
Almost all the work is covered by https://github.com/pytorch/pytorch/pull/126578

Once that PR is merged, I will rebase the PR to just get the tests in.

**Review by yf225 — APPROVED** [review]

**Comment by anijain2305** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/129045

**Pull request closed without merging**